src/json.c

```diff
@@ -214,11 +284,20 @@ int JSON_StoreInDocField(RedisJSON json, JSONType jsonType, FieldSpec *fs, struc
     case JSONType_Null:
       df->unionType = FLD_VAR_T_NULL;
       break;
-    case JSONType_Array:;
-      // Flattening the array to go over it with iterator api
-      JSONResultsIterator arrIter = japi->get(json, "$.[*]");
-      rv = JSON_StoreVectorInDocField(fs, arrIter, df);
-      japi->freeIter(arrIter);
+    case JSONType_Array:
+      if (fs->types == INDEXFLD_T_FULLTEXT || fs->types == INDEXFLD_T_VECTOR) {
+        // Flattening the array to go over it with iterator api
+        // (using a path which is rooted at the current array)
+        JSONResultsIterator arrIter = japi->get(json, "$.[*]");
```

Comment:
Wouldn't it be more efficient to just iterate the array ourself instead of using jsonpath that need to be parsed and executed?

Reply:
Sure it would be more efficient to avoid the parse/select,
But I assumed it would be rather negligible, and would allow simple code reuse,
and if improvement is needed, can be done in a later phase.
(maybe add an API such as `JSONResultsIterator (*getIter)(RedisJSON json)`)

WDYT?

Reply:
But you have the length of the array, and you can get element but index, it should be a simple loop no?